src/LearnositySdk/Utils/Json.php

```diff
@@ -104,4 +111,67 @@ public static function isJson($val)
     {
         return !is_null(json_decode($val, true));
     }
+
+    /**
+     * This function builds up a map of float values in scientific notation and the equivalent float
+     * notations. json_encode encodes float values in scientific notation, but we want to make sure
+     * they are in float notation
+     *
+     * @param $json the $json to be encoded
+     * @param $floatMap pointer to the floatMap
+     */
+    private static function getHighPrecisionFloatMap($json, &$floatMap)
+    {
+        if (is_array($json)) {
+            foreach ($json as $value) {
+                if (is_array($value)) {
+                    self::getHighPrecisionFloatMap($value, $floatMap);
+                }
+                if (is_float($value)) {
+                    $currentMap = self::getFloatMap($value);
+                    if ($currentMap) {
+                        $floatMap[] = $currentMap;
+                    }
+                }
+            }
+        } elseif (is_float($json)) {
+            $currentMap = self::getFloatMap($json);
+            if ($currentMap) {
+                $floatMap[] = $currentMap;
+            }
+        }
+    }
+
+    /**
+     * Helper function to build up the float map
+     *
+     * @param $value
+     * @return array|bool
+     */
+    private static function getFloatMap($value)
+    {
+        $stringValue = json_encode($value);
```

Comment:
Should we check that the $value is a `float` here, and otherwise return before doing anything?

Reply:
It's checked in the calling function, so I think it's not necessary

Reply:
Actually I've added a type hint